docs/deployment/monitoring.md

```diff
@@ -0,0 +1,27 @@
+# Monitoring
+
+UniCore offers a variety of monitoring options to ensure a healthy deployment and successful operations.
+
+## Probes
+
+UniCore implements conventional probe endpoints to monitor the availability and health of the service. [Standard Kubernetes probes](https://kubernetes.io/docs/concepts/configuration/liveness-readiness-startup-probes/) are served at the following endpoints:
+
+- `/healthz`: Liveness probe
+<!-- - `/readyz`: Readiness probe -->
+
+## Metadata
+
+The following endpoints provide metadata about the service itself and allow for more detailed deployment monitoring:
+
+- `/version`: Returns the version and the git commit hash
+- `/info`: Returns the version, the git commit hash, a Docker build timestamp and the container uptime
```

Comment:
I'm just wondering, so these endpoints seem to be semi-conventional, but there doesn't seem to be any specification that specifies how they should be implemented (afaik?). So did you come up yourself with this format:
```json
{
    "version": "feat-version",
    "git_commit_hash": "18f26e3",
    "docker_build_timestamp": "2025-03-14T12:46:57.821Z",
    "uptime": "0 days, 00:29:38"
}
```
Should/could we follow any convention here? (I don't know I'm just interested :))

Reply:
There is no _actual_ convention for it, as far as I know. Spring Boot (a widely-used Java framework) has a [similar feature](https://www.baeldung.com/spring-boot-info-actuator-custom), but the keys are non-conventional. You can basically put anything there that you think could be interesting for a developer/operator.

Reply:
Yeah `revision` is nice, but `git_commit_hash` works as well imo :+1:

Reply:
The whole point of this endpoint is to give an operator insight into what version of the application is running on that URL. So it should be human-readable and not too technical and/or nested. While `rev` or `revision` would be more consistent with the style cargo dependencies are defined, it's less apparent to an operator (that probably doesn't even know Rust). Since we're in the lead and there's no actual convention for this endpoint, I'm fine with both `revision` (more technical) or `git_commit_hash` (more human-readable).